packages/ThreadPool/test/hhpccg/CMakeLists.txt

```diff
@@ -27,7 +27,7 @@ TRIBITS_ADD_EXECUTABLE(
   test_tpi_hhpccg
   COMM serial mpi
   SOURCES main.c CGSolver.c BoxPartitionIB.c tpi_vector.c dcrs_matrix.c
-  DEPLIBS pthread m
```

Comment:
Actually, `pthread` should be pulled in as the "Pthread" TPL defined in Trilinos/TPLsList.cmake and declared as an optional dependency in ThreadPool/cmake/Dependencies.cmake.   These tests should only get created if `${PACKAGE_NAME}_ENABLE_Pthread` is true.  You can address this by replacing `DEPSLIBS pthread m` with:

```
  TRIBITS_ADD_EXECUTABLE(
    ...
    EXCLUDE_IF_NOT_TRUE  ${PACKAGE_NAME}_ENABLE_Pthread
    IMPORTEDLIBS m
    ...
    )
```

As the documentation for `IMPORTEDLIBS` says:

* https://tribits.org/doc/TribitsDevelopersGuide.html#formal-arguments-tribits-add-executable

"So far, the only case where `IMPORTEDLIBS` has been shown to be necessary is to pass in the standard C math library `m`.  In every other case, a TriBITS TPL should be used instead."

Does that make sense?